### PR TITLE
Address SPJ's comments

### PR DESCRIPTION
**Pull request by rohanjr**
Main diff is in Proposed Change section. Makes "layout herald" terminology and behaviour explicit and includes BNF grammar extending the Haskell 2010 spec.

### CODE REVIEW

File: proposals/0000-record-with-syntax.rst [hunk 1]
```diff
@@ -223,6 +246,236 @@ The following example shows a GADT declaration using ``with`` syntax.
 
 Note that to write ``Con2`` on a single line, the curly braces are required to separate the field type from the result type.
 
+Extension to BNF Grammar
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here we show the proposed extension to the BNF grammar of the Haskell 2010 base language.
+We have taken the current `grammar <https://www.haskell.org/onlinereport/haskell2010/haskellch10.html#x17-18000010.5>`_
+and added lines marked with a ``-- NEW`` comment.
```
Comment (ghost) [inline proposals/0000-record-with-syntax.rst]: I'm not sure you need to provide the whole grammar? Just the `aexp` production should do?

File: proposals/0000-record-with-syntax.rst [hunk 2]
```diff
@@ -223,6 +246,236 @@ The following example shows a GADT declaration using ``with`` syntax.
...
+  		|	( exp )	    (parenthesized expression)
+  		|	( exp1 , … , expk )	    (tuple, k ≥ 2)
+  		|	[ exp1 , … , expk ]	    (list, k ≥ 1)
+  		|	[ exp1 [, exp2] .. [exp3] ]	    (arithmetic sequence)
+  		|	[ exp | qual1 , … , qualn ]	    (list comprehension, n ≥ 1)
+  		|	( infixexp qop )	    (left section)
+  		|	( qop⟨-⟩ infixexp )	    (right section)
+  		|	qcon { fbind1 , … , fbindn }		(labeled construction, n ≥ 0)
+  		|	qcon { fbind1 ; … ; fbindn }		(labeled construction, n ≥ 0) -- NEW
+  		|	qcon with { fbind1 ; … ; fbindn }	(labeled construction, n ≥ 0) -- NEW
```
Comment (ghost) [inline proposals/0000-record-with-syntax.rst]: I think you can collapse these two cases by writing `[with]` (to indicate 0 or 1 occurrences).

File: proposals/0000-record-with-syntax.rst [hunk 3]
```diff
@@ -223,6 +246,236 @@ The following example shows a GADT declaration using ``with`` syntax.
...
+  		|	aexp⟨qcon⟩ { fbind1 , … , fbindn }	(labeled update, n  ≥  1)
+  		|	aexp⟨qcon⟩ { fbind1 ; … ; fbindn }	(labeled update, n  ≥  1) -- NEW
```
Comment (ghost) [inline proposals/0000-record-with-syntax.rst]: Ditto.

File: proposals/0000-record-with-syntax.rst [hunk 4]
```diff
@@ -356,13 +607,14 @@ No unresolved questions.
...
 We do not require any changes to GHC in advance of the feature.
 
 The implementation entails the following changes to the compiler frontend:
 
 1. Extra ``with`` token in the lexer (conditional on the ``RecordWith`` extension being set), enabling layout syntax
-2. Extra productions in the parser for ``with`` followed by record fields within virtual braces, storing source locations as usual
-3. Boolean flags in AST nodes involving records, indicating if ``with`` syntax was used
-4. Embellishment of pretty printers to respect layout syntax (e.g. when reporting error messages)
+2. Extra parser rules for ``with`` followed by record fields within virtual braces
+3. Modified parser rules for records without ``with`` using semicolons instead of commas
```
Comment (ghost) [inline proposals/0000-record-with-syntax.rst]: These are new, not modified rules no?

**Review by ghost — APPROVED** [review]
Nice. Couple of inline comments.

File: proposals/0000-record-with-syntax.rst (hunk 4 above)
Comment (rohanjr-da) [reply proposals/0000-record-with-syntax.rst]: True, will adjust

File: proposals/0000-record-with-syntax.rst (hunk 2 above)
Comment (rohanjr-da) [reply proposals/0000-record-with-syntax.rst]: Ah, nice! Will do

File: proposals/0000-record-with-syntax.rst (hunk 1 above)
Comment (rohanjr-da) [reply proposals/0000-record-with-syntax.rst]: Good point, I could just include `constr`, `aexp`, and `apat`.

**Pull request merged**